container/go/pkg/utils/utils.go

```diff
@@ -44,3 +49,49 @@ func ReadImage(src, format string) (v1.Image, error) {
 
 	return nil, errors.Errorf("unknown image format %q", format)
 }
+
+// TODO: REMOVE these two functions copied from Winnie's PR, refactor after her PR is merged.
+type formattedString map[string]interface{}
+
+// formateWithMap takes all variables of format {{.VAR}} in the input string `format`
+// and replaces it according to the map of parameters to values in `params`.
+func formatWithMap(format string, params formattedString) string {
+	msg := &bytes.Buffer{}
+	template.Must(template.New("").Parse(format)).Execute(msg, params)
+	return msg.String()
+}
+
+// Stamp provides the substitutions of variables inside {} using info in file pointed to
+// by stampInfoFile.
+func Stamp(inp string, stampInfoFile []string) (string, error) {
```

Comment:
The implementation of this function is too complex imo. I think you should be able to do just the following:
Read the stamp file line by line as you are doing. Then, instead of using templates which is overkill here, simply replace all occurrences of `"{" + key + "}"` with the stamp value in the given input string.

Also, it might be a good idea to create a stamper struct that is initialized with a list of stamp files. The stamper would load all the files into maps of key value pairs. This stamper could then be given values to stamp and it would go through the pre-processed maps and replace any keys it finds. This way, you avoid read the stamp files for *every* value.